src/systemcmds/param/param.cpp

```diff
@@ -216,12 +221,21 @@ param_main(int argc, char *argv[])
 		}
 
 		if (!strcmp(argv[1], "import")) {
+			int ret;
 			if (argc >= 3) {
-				return do_import(argv[2]);
+				ret = do_import(argv[2]);
 
 			} else {
-				return do_import();
+				ret = do_import();
+			}
+
+#if defined (__PX4_NUTTX) && !defined (CONFIG_BUILD_FLAT) && !defined(__KERNEL__)
+			if (ret == 0) {
+				argv[0] = (char *)kparam_main_fn;
```

Comment:
This is a bit awkward. Let me review the param and kernel param situation and see what else we can do.

We don't want to have to spend the next few years gradually cleaning up these CONFIG_BUILD_FLAT ifdefs.

Reply:
Agreed.

Again part of the thing that I wanted to do a top-down approach. First set up the framework and a working setup for a protected target, and later dig in to individual modules to clean up the interfaces. CONFIG_BUILD_FLAT ifdefs are found in those components which are on the userspace/kernelspace interface border.

There is always an option of putting these things in another source file, and adding a function call here, which is then stubbed away when doing normal "flat" build. I will just look into this again

Reply:
I moved all of this first into just one place in this file. Further it can be cleaned up by leving just a declaration into this file, and put the code in another source file? But having it in this same file is least amount of changes.